File: app/src/main/java/com/zxcx/zhizhe/ui/classify/ClassifyCardBagBean.java
package com.zxcx.zhizhe.ui.classify;

import com.chad.library.adapter.base.entity.MultiItemEntity;
import com.google.gson.annotations.SerializedName;
import com.zxcx.zhizhe.retrofit.RetrofitBaen;

/**
 * Created by anm on 2017/8/30.
 */

public class ClassifyCardBagBean extends RetrofitBaen implements MultiItemEntity{

    public static final int TYPE_CARD_BAG = 2;

    @SerializedName("id")
    private int id;
    @SerializedName("titleImage")
    private String imageUrl;
    @SerializedName("title")
    private String name;

    @Override
    public int getItemType() {
        return TYPE_CARD_BAG;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: app/src/main/java/com/zxcx/zhizhe/ui/classify/ClassifyCardBagBean.kt
package com.zxcx.zhizhe.ui.classify

import com.chad.library.adapter.base.entity.MultiItemEntity
import com.google.gson.annotations.SerializedName
import com.zxcx.zhizhe.retrofit.RetrofitBaen

/**
 * Created by anm on 2017/8/30.
 */

class ClassifyCardBagBean : RetrofitBaen(), MultiItemEntity {

    @SerializedName("interested")
    var isChecked: Boolean = false
    @SerializedName("id")
    var id: Int = 0
    @SerializedName("titleImage")
    var imageUrl: String? = null
    @SerializedName("title")
    var name: String? = null

    override fun getItemType(): Int {
        return TYPE_CARD_BAG
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as ClassifyCardBagBean

        if (id != other.id) return false

        return true
    }

    override fun hashCode(): Int {
        return id
    }

    companion object {

        const val TYPE_CARD_BAG = 2
    }
}
